crate/operator/create.py

```diff
@@ -155,22 +156,41 @@ async def create_debug_volume(
 def get_sql_exporter_config(
     owner_references: Optional[List[V1OwnerReference]], name: str, labels: LabelType
 ) -> V1ConfigMap:
+
     sql_exporter_config = pkgutil.get_data("crate.operator", "data/sql-exporter.yaml")
-    responsivity_collector_config = pkgutil.get_data(
-        "crate.operator", "data/responsivity-collector.yaml"
-    )
-    if sql_exporter_config and responsivity_collector_config:
-        return V1ConfigMap(
+
+    if sql_exporter_config:
+        # Parse the config yaml file to get the defined collectors and load them
+        parsed_sql_exporter_config = yaml.load(sql_exporter_config.decode())
+        collectors = parsed_sql_exporter_config["target"]["collectors"]
+
+        result = V1ConfigMap(
             metadata=V1ObjectMeta(
                 name=f"crate-sql-exporter-{name}",
                 labels=labels,
                 owner_references=owner_references,
             ),
             data={
                 "sql-exporter.yaml": sql_exporter_config.decode(),
-                "responsivity-collector.yaml": responsivity_collector_config.decode(),
             },
         )
+
+        # Add the yaml collectors to the configmap dynamically
+        for collector in collectors:
+            if collector.endswith("_collector"):
+                collector = collector[:-10]
```

Comment:
alternatively, for better readability
```suggestion
                collector = collector.replace("_collector", "")
```

Reply:
I would rather add a comment, since the proposed change does not guarantee it will delete only the suffix. I'm probably being too picky on this one though!